### PR TITLE
working app listing check

### PR DESCRIPTION
**Pull request by aaustin**
@qinweigong 

This works differently than the Android version. Instead of harvesting a list, it will check an existing list of URI schemes that we have in the database. Currently, it's a very short and simple list stored in the API. We'll use the Android harvesting to accumulate a rich data set of URI schemes

### CODE REVIEW

File: Branch-SDK/Branch-SDK/BNCConfig.h [hunk 1]
```diff
@@ -24,7 +24,7 @@
 #endif
 
 #ifdef BNC_DEV_ENV
-#define BNC_API_BASE_URL        @"http://localhost:3001"
+#define BNC_API_BASE_URL        @"http://localhost:3000"
```
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/BNCConfig.h]: so dashboard is using port 3001?

File: Branch-SDK/Branch-SDK/Branch.m [hunk 2]
```diff
@@ -1208,7 +1236,9 @@ - (void)serverCallback:(BNCServerResponse *)response {
                     [BNCPreferenceHelper setInstallParams:[response.data objectForKey:DATA]];
                 }
             }
-            
+            if ([BNCPreferenceHelper getNeedAppListCheck]) {
+                [self processListOfApps];
+            }
```
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: why do we need to call this again here? (same as line 1210-1212?)

File: Branch-SDK/Branch-SDK/BNCConfig.h (hunk 1 above)
Comment (aaustin) [inline Branch-SDK/Branch-SDK/BNCConfig.h]: I just use localhost 3000 for my API when I'm testing. No big deal if you prefer to keep it at 3001

File: Branch-SDK/Branch-SDK/Branch.m (hunk 2 above)
Comment (aaustin) [inline Branch-SDK/Branch-SDK/Branch.m]: It gets checked and added in both 'install' and 'open' callbacks, as scans the apps every 4 days or so

Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: Got it!

**Comment by qinweigong** [comment]
:+1:

**Pull request closed without merging**